src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs

```diff
@@ -100,16 +100,24 @@ public async Task SynchronizeWithServer()
             {
                 var projectKey = bindingConfiguration.Project.ProjectKey;
                 var taintVulnerabilities = await sonarQubeService.GetTaintVulnerabilitiesAsync(projectKey, CancellationToken.None);
-
+                
                 logger.WriteLine(TaintResources.Synchronizer_NumberOfServerIssues, taintVulnerabilities.Count);
 
-                var taintIssueVizs = taintVulnerabilities.Select(converter.Convert).ToArray();
-                taintStore.Set(taintIssueVizs);
+                var hasTaintIssues = taintVulnerabilities.Count > 0;
 
-                var hasTaintIssues = taintIssueVizs.Length > 0;
-                UpdateTaintIssuesUIContext(hasTaintIssues);
-                if (hasTaintIssues)
+                if (!hasTaintIssues)
+                {
+                    ClearStore();
+                    UpdateTaintIssuesUIContext(false);
+                }
+                else
                 {
+                    var analysisInformation = await GetAnalysisInformation(projectKey);
```

Comment:
Fetching the project branches only if we really have issues.
If we don't, then we won't show the taint window anyway, and it will contain the empty overlay. So no point in fetching the information.

Reply:
The user can still manually open the window. What will it show in that case?

Reply:
No, it will contain an empty overlay.